File: tests/fixtures/edgecase_repo/scala/Service3.scala
package com.example.gen

import com.example.gen.Base3

trait Handler3 {
  def handle(payload: String): Boolean
}

class Base3(val name: String)

class Service3(name: String) extends Base3(name) with Handler3 {
  def handle(payload: String): Boolean = {
    this.validate(payload)
    true
  }

  private def validate(payload: String): Unit = {}
}

object Helpers3 {
  def helper(value: String): String = value
}

File: tests/fixtures/edgecase_repo/swift/Service3.swift
import Foundation

protocol Handler3 {
    func handle(_ payload: String) -> Bool
}

class Base3 {
    let name: String
    init(name: String) {
        self.name = name
    }
}

class Service3: Base3, Handler3 {
    func handle(_ payload: String) -> Bool {
        self.validate(payload)
        return true
    }

    private func validate(_ payload: String) {}
}

func helper3(_ value: String) -> String {
    return value
}
